bitcoin/src/blockdata/script.rs

```diff
@@ -400,6 +400,28 @@ impl Script {
       Builder::new()
     }
 
+    /// Returns the public key if this script is P2PK with a valid public key.
+    pub fn p2pk_public_key(&self) -> Option<PublicKey> {
+        if self.is_p2pk() {
+            PublicKey::from_slice(self.p2pk_pubkey_bytes()?).ok()
+        } else {
+            None
+        }
+    }
```

Comment:
Since this now accepts scripts that don't have `OP_PUSHBYTES` or `OP_CHECKSIG` and you didn't catch it I suggest adding tests for scripts missing those. :)

Reply:
Good catch, added